### PR TITLE
[Continous Batching] fix do_Sample=True in continuous batching

### PR DESCRIPTION
**Pull request by kashif**
This pull request introduces improvements to sampling support in continuous batching generation, enhances tensor shape handling for logit processors, and expands test coverage for batch generation with sampling. The main changes are grouped below:

**Sampling and Generation Behavior:**

* Enabled sampling in `batch_generate` by setting `do_sample=True` and configuring `temperature` and `top_p` parameters, allowing for more diverse outputs during batch generation.

**Tensor Shape Handling and Sampling Logic:**

* Updated the logit processor in `continuous_api.py` to handle 3D logits produced by continuous batching, reshaping them as needed for compatibility with logit processors and restoring original shapes after processing.
* Improved sampling logic in `_sample` to ensure correct tensor dimensions when sampling tokens, making the method robust for both sampling and greedy decoding.

**Paged Attention and Integration:**

* Added a check for the presence of `flash_attn_varlen_func` in the attention implementation to ensure compatibility and prevent errors during paged attention forward passes.

**Testing and Validation:**

* Refactored batch output decoding in tests to use `generated_tokens` instead of `static_outputs` for accuracy.
* Added a new parameterized test `test_generate_batch_with_sampling` to verify that batch generation with sampling enabled works correctly across different attention implementations and configurations. The test checks for successful completion, non-empty outputs, and actual sampling behavior.

### CODE REVIEW

**Review by McPatate — APPROVED** [review]
lgtm, ty!

File: examples/pytorch/continuous_batching.py [hunk 1]
```diff
@@ -229,7 +229,9 @@ def batch_generate(
         use_cuda_graph=args.use_cuda_graph,
         eos_token_id=tokenizer.eos_token_id,
         pad_token_id=tokenizer.pad_token_id,
-        do_sample=False,
+        do_sample=True,
+        temperature=0.8,
+        top_p=1.0,
```
Comment (kashif) [inline examples/pytorch/continuous_batching.py]:
```suggestion
        top_p=0.9,
```

**Review by ArthurZucker — APPROVED** [review]
thanks! perfect 🤗
